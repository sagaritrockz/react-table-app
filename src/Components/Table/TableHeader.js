import React from "react";

const TableHeader = ({ header }) => {
    return (<th>{header}</th>);
};

TableHeader.defaultProps = {
    header: ''
};

export default TableHeader;
