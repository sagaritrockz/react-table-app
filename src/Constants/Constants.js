const API = "https://api.worldbank.org/v2/countries/";

const FILTERS = {
    region: [
        {id: "LCN", iso2code: "ZJ", value: "Latin America & Caribbean"},
        {id: "SAS", iso2code: "8S", value: "South Asia"},
        {id: "NA", iso2code: "NA", value: "Aggregates"},
        {id: "SSF", iso2code: "ZG", value: "Sub-Saharan Africa "},
        {id: "ECS", iso2code: "Z7", value: "Europe & Central Asia"},
        {id: "MEA", iso2code: "ZQ", value: "Middle East & North Africa"},
        {id: "EAS", iso2code: "Z4", value: "East Asia & Pacific"},
        {id: "NAC", iso2code: "XU", value: "North America"}
    ],
    incomeLevel: [
        {id: "HIC", iso2code: "XD", value: "High income"},
        {id: "INX", iso2code: "XY", value: "Not classified"},
        {id: "LIC", iso2code: "XM", value: "Low income"},
        {id: "LMC", iso2code: "XN", value: "Lower middle income"},
        {id: "LMY", iso2code: "XO", value: "Low & middle income"},
        {id: "MIC", iso2code: "XP", value: "Middle income"},
        {id: "UMC", iso2code: "XT", value: "Upper middle income"}
    ],
    lendingType: [
        {id: "IBD", iso2code: "XF", value: "IBRD"},
        {id: "IDB", iso2code: "XH", value: "Blend"},
        {id: "IDX", iso2code: "XI", value: "IDA"},
        {id: "LNX", iso2code: "XX", value: "Not classified"}
    ]
}

const HEADERS = ['name', 'iso2Code', 'region', 'capitalCity', 'lendingType', 'incomeLevel']

export { FILTERS, HEADERS, API };
