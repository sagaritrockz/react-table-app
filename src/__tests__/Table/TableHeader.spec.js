import React from 'react'
import Enzyme from 'enzyme'
import Adaptar from 'enzyme-adapter-react-16.3'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import TableHeader from '../../Components/Table/TableHeader'
import { HEADERS } from '../../Constants/Constants'

Enzyme.configure({adapter: new Adaptar()});

describe('TableHeader', () => {
    it('should render correctly', () => {      
        const output = shallow(
            <TableHeader headers={HEADERS} />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});