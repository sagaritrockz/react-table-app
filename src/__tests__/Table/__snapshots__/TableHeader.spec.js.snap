// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader should render correctly 1`] = `
Array [
  <th
    key="0"
>
    Name
</th>,
  <th
    key="1"
>
    Iso2 Code
</th>,
  <th
    key="2"
>
    Region
</th>,
  <th
    key="3"
>
    Capital City
</th>,
  <th
    key="4"
>
    Lending Type
</th>,
  <th
    key="5"
>
    Income Level
</th>,
]
`;
